src/boost_histogram/_internal/hist.py

```diff
@@ -597,8 +597,13 @@ def _axis(self, i: int = 0) -> Axis:
     def storage_type(self) -> Type[Storage]:
         return cast(self, self._hist._storage_type, Storage)  # type: ignore[return-value]
 
-    # Backward compat
-    _storage_type = storage_type
+    @property
+    def _storage_type(self) -> Type[Storage]:
+        warnings.warn(
+            "Accessing storage type has changed from _storage_type to storage_type, and will be removed in future.",
+            DeprecationWarning,
```

Comment:
```suggestion
            DeprecationWarning,
            stacklevel=2,
```

It is virtually never correct to call `warnings.warn` without `stacklevel` of at least 2 - this means _using_ `_storage_type` is deprecated.